src/hid/hid-device.cpp

```diff
@@ -195,7 +215,27 @@ namespace librealsense
         {
             REALSENSE_HID_REPORT report;
 #ifdef __APPLE__
-            hid_read(_hidapi_device, reinterpret_cast<unsigned char*>(&report), sizeof(REALSENSE_HID_REPORT));
+            unsigned char tmp_buffer[100] = { 0 };
+            hid_read( _hidapi_device, tmp_buffer, _realsense_hid_report_actual_size );
+            if( _realsense_hid_report_actual_size != sizeof( REALSENSE_HID_REPORT ) )
+            {
+                // for FX version<5.16 the actual struct is 32 bit, so we can not use memcpy for the whole struct
+                // for FX version<5.16 x,y,x are all short with the x variable located at offset 10, y at offset 12, and
+                // z at offset 14 within the structure.
+                memcpy( &report, tmp_buffer, 10 );
+                const int16_t * x = reinterpret_cast< const int16_t * >( tmp_buffer + 10 );
+                const int16_t * y = reinterpret_cast< const int16_t * >( tmp_buffer + 12 );
+                const int16_t * z = reinterpret_cast< const int16_t * >( tmp_buffer + 14 );
+                report.x = *x;
```

Comment:
We can do lines 226-228 directly on lines 229-231 right?
If you change please make sure it still works :)

Reply:
we can but we wanted to make it more readable and simpler to understand.